app/controllers/supplemental_files_controller.rb

```diff
@@ -136,6 +136,34 @@ def edit_structure_path
       edit_media_object_path(media_object_id, step: 'file-upload')
     end
 
+    def edit_file_information
+      ident = "machine_generated_#{params[:id].to_s}".to_sym
+
+      if params[ident] && !@supplemental_file.machine_generated?
+        @supplemental_file.tags += ['machine_generated']
+        edit_file_label supplemental_file_params[:label]
+      elsif !params[ident] && @supplemental_file.machine_generated?
+        @supplemental_file.tags -= ['machine_generated']
+        label = supplemental_file_params[:label].gsub(" (machine generated)", '')
+        @supplemental_file.label = label
+      elsif params[ident] && @supplemental_file.machine_generated?
+        edit_file_label supplemental_file_params[:label]
+      else
+        @supplemental_file.label = supplemental_file_params[:label]
+      end
+    end
+
+    def edit_file_label label
+      label = if label.exclude?('(machine generated)') && File.extname(label).present?
+                label.gsub(File.extname(label), " (machine generated)#{File.extname(label)}")
+              elsif label.exclude?('(machine generated)') && File.extname(label).blank?
+                label + (" (machine generated)")
+              else
+                label
+              end
+      @supplemental_file.label = label
+    end
```

Comment:
When a user downloads a transcript file from the transcript viewer, the filename of the downloaded file is the label value. This adds "(machine generated)" to the label so that the download includes it. 

Having the label changed on the database end causes some minor issue with the frontend where there is some Javascript to update the label on save. This could cause confusion for users who save a label and then next time they reload the page, the label is different and includes information they had added themselves. I looked over the JS but could not figure out the best approach for getting things to generate on the frontend properly to replicate current behavior.

Since the transcript viewer will eventually have a message about being machine generated, do we need to change the label here or is there somewhere in the Ramp code where the change to the label/filename could be made as the download occurs? Would that be a better approach?